operator.go

```diff
@@ -6,10 +6,21 @@ import (
 	"text/template"
 )
 
+type encoding string
```

Comment:
What do you think about this: https://github.com/giantswarm/architect/blob/master/configuration/provider/provider.go? 

Having constants with the specific private type allow decent godoc and easy use out of the box. IMO we do not need a String() method, we only need the allowed options protected by a package type. That also prevents calling some method with empty strings like `encoding("")`.